Braze-Demo-Match-Game-Content-Extension/ViewController/NotificationViewController.swift

```diff
@@ -0,0 +1,185 @@
+  
+  // MARK: - Outlets
+  @IBOutlet private weak var stackView: UIStackView!
+  @IBOutlet private weak var playAgainButton: UIButton!
+  @IBOutlet private weak var bestScoreLabel: UILabel!
+  @IBOutlet private weak var attemptsLabel: UILabel!
+  
+  // MARK: - Actions
+  @IBAction func playAgainPressed(_ sender: UIButton) {
+    resetGame()
+  }
+  
+  // MARK: - Variables
+  private var matchGame = MatchGame()
+  private var cardViews = [MatchCardView]()
+  
+  private var boardLayout: (rows: Int, columns: Int) {
+    if traitCollection.verticalSizeClass == .compact {
+      return (2, 4)
+    } else {
+      return (3, 4)
+    }
+  }
+  
+  // lock/unlock the board to avoid fast fingers, no more than 2 cards visible at a time
+  private var lockBoard = false {
+    didSet {
+     view.isUserInteractionEnabled = !lockBoard
+    }
+  }
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    
+    configureBoard(rows: boardLayout.rows, columns: boardLayout.columns)
+    matchGame.configureGame(numberOfCards: cardViews.count, delegate: self)
+  }
+  
+  func didReceive(_ notification: UNNotification) {
+    // do nothing
+  }
+}
+
+// MARK: - MatchGame Delegate
+extension NotificationViewController: MatchGameDelegate {
+  func cardsDidLoad(_ cards: [MatchCard]) {
+    configureCardViews(cards)
+  }
+  
+  func cardsDidMatch(_ indicies: [Int], currentScore: Int) {
+    updateBoard(currentScore: currentScore) {
+      self.fadeOutMatchedCards(at: indicies)
+    }
+  }
+  
+  func cardsDidNotMatch(_ indicies: [Int], currentScore: Int) {
+    updateBoard(currentScore: currentScore) {
+      self.flipUnmatchedCards(at: indicies)
+    }
+  }
+}
+
+// MARK: - Private Methods
+private extension NotificationViewController {
+  func configureBoard(rows: Int, columns: Int) {
+    for _ in 0..<rows {
+      let matchCardsStackView = UIStackView()
+      matchCardsStackView.axis = .horizontal
+      matchCardsStackView.distribution = .equalSpacing
+      matchCardsStackView.spacing = 10
+
+      for _ in 0..<columns {
+        let cardView: MatchCardView = .fromNib()
+        
+        matchCardsStackView.addArrangedSubview(cardView)
+        cardView.translatesAutoresizingMaskIntoConstraints = false
+        cardView.widthAnchor.constraint(equalTo: cardView.heightAnchor, multiplier: 2.5/3.5).isActive = true
+        
+        cardViews.append(cardView)
+      }
+      
+      stackView.addArrangedSubview(matchCardsStackView)
+    }
+  }
+  
+  /// Configures each `MatchCardView`'s flipped image with an associated `selectedImage` (torchie) and `tag`. If the number of `cards` do not equal the number of `cardViews`, mistakes were made.
+  /// - parameter cards: The array of shuffled cards that will be assigned to an array of `cardViews` one by one.
+  func configureCardViews(_ cards: [MatchCard]) {
+    guard cards.count == cardViews.count else { fatalError("Mistakes were made") }
```

Comment:
Does this crash it?

Reply:
Yup! For the content extension if it crashes it'll just display a blank view. I suppose there's a better way to handle that error like display a view that says dismiss the push or something but it's just an assurance that everything is synced. 

<img src=https://user-images.githubusercontent.com/5905170/108891251-1d4c7a80-75d4-11eb-9c00-da0caa7f7584.png width=414 height=896>